src/components/tl/ucp/barrier/barrier.h

```diff
@@ -18,4 +18,7 @@ extern ucc_base_coll_alg_info_t
 
 ucc_status_t ucc_tl_ucp_barrier_init(ucc_tl_ucp_task_t *task);
 
+ucc_status_t ucc_tl_ucp_barrier_knomial_start(ucc_coll_task_t *task);
+void         ucc_tl_ucp_barrier_knomial_progress(ucc_coll_task_t *task);
```

Comment:
These function declarations were previously in barrier.c. I moved them to barrier.h so that they may be included and called from the sliding window algorithm.

Reply:
ok, barrier_knomial_init you need to initialize task, but why do you need start and progress?

Reply:
this was so that i can call them once the allgather and puts/gets are done.